### PR TITLE
Brave VPN desktop service status handling

### PR DESCRIPTION
**Pull request by simonhong**
Whenever connect is asked, vpn service tries to create os vpn entry
with latest hostname and user credentials from guardian service.

fix brave/brave-browser#18648
fix brave/brave-browser#18422

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`npm run test brave_unit_tests -- --filter=*VPN*`

1. Launch brave and enable vpn feature from brave://flags
2. Close and launch again with `--brave-vpn-test-monthly-pass=xxxxxx` testable pass
3. Click vpn button and launch vpn panel and try connect/disconnect and change region

### CODE REVIEW

File: components/brave_vpn/resources/panel/container.tsx [hunk 1]
```diff
@@ -80,9 +80,10 @@ function Main () {
       onConnectionStateChanged: (state: ConnectionState) => {
         if (state === ConnectionState.CONNECT_FAILED) {
           setHasError(true)
-          setOn(false)
         }
 
+        setOn(state == ConnectionState.CONNECTED ||
+              state === ConnectionState.CONNECTING);
```
Comment (nullhook) [inline components/brave_vpn/resources/panel/container.tsx:86]: Why does the first one has two equal operator vs the second one with three?

Comment (simonhong) [reply components/brave_vpn/resources/panel/container.tsx:86]: it's mistake :)

File: components/skus/browser/skus_utils.cc [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/skus/browser/skus_utils.h"
+
+#include "base/command_line.h"
+#include "base/notreached.h"
```
Comment (bsclifton) [inline components/skus/browser/skus_utils.cc:9]: I think this include can be removed

File: components/brave_vpn/brave_vpn_service_desktop.cc [hunk 3]
```diff
@@ -246,12 +402,60 @@ void BraveVpnServiceDesktop::LoadCachedRegionData() {
...
+      VLOG(2) << __func__ << " : "
+              << "Loaded cached device region";
+    }
+  }
+}
+
+void BraveVpnServiceDesktop::LoadPurchasedState() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  auto* cmd = base::CommandLine::ForCurrentProcess();
+  if (cmd->HasSwitch(brave_vpn::switches::kBraveVPNTestMonthlyPass)) {
```
[the remark this answers is not in the document]
Comment (bsclifton) [reply components/brave_vpn/brave_vpn_service_desktop.cc:422]: OMG - yeah I forgot about macOS use case. For unsigned builds, it's still OK 😄 I think we just don't want it on official builds (ones we push out the users)

**Review by bsclifton — APPROVED** [review]
Nice work! And thanks for adding the GetEnvironment 😄👍

File: components/skus/browser/skus_utils.cc (hunk 2 above)
Comment (simonhong) [reply components/skus/browser/skus_utils.cc]: Deleted.

**Comment by simonhong** [comment]
Merged with skip because last change was deleting simple unused header including.

**Pull request merged**